### DISCUSSION TITLE
Any reason my laptop suddenly says it contains a virus after having used the program for months now?

### DISCUSSION THREAD

**Discussion by DingleDangleNootNoot** [valinet/ExplorerPatcher#2874] [Q&A] [answered]
![Screenshot 2024-02-20 142017](https://github.com/valinet/ExplorerPatcher/assets/22161807/04ddc02c-228d-45c5-be09-c358866fe177)

Been using this for a while now and only now does it think it's a virus and won't update. Anything I should know or do to fix this?

**Comment by Mersadon** [comment]
Few minutes ago, my Windows Defender identified ExplorerPatcher to be a "high threat" and put it into quarantine. 

I am using ExplorerPatcher for over a year now, and this never happened. Not sure what is the reason, perhaps some database has been updated somewhere... ?

![ExplorerPatcher Windows Defender](https://github.com/valinet/ExplorerPatcher/assets/39765904/c99f8de3-5aac-4092-9f20-4e4cba3aa5d7)

**Reply by eduardopatricio** [reply]
I tried to install ExplorerPatcher for the first time just now and Windows Defender removed it right  away. 
![image](https://github.com/valinet/ExplorerPatcher/assets/118542/9bf896e0-f923-4db5-87b4-aaf7a11a1b92)

**Reply by Mersadon** [reply]
Yeah, well on their website Microsoft says that it "might" be a threat, not that it "is" a threat. 
I guess it is a false positive, because Microsoft does not like you messing around with W11 (well, too bad for them and their crappy OS 😛).

![hacktools](https://github.com/valinet/ExplorerPatcher/assets/39765904/c8034b23-eb17-48de-a82f-2150fb7c11b3)

**Reply by cmartinnyc** [reply]
Having same issues here. Have been using this program for months and all of a sudden getting trojans popping up on my system.

Any input here from the developer please?

![Screenshot 2024-02-21 222410](https://github.com/valinet/ExplorerPatcher/assets/160811362/e168d4b8-102b-41e9-bf6a-bdcbe3b918a2)
![Screenshot 2024-02-21 222433](https://github.com/valinet/ExplorerPatcher/assets/160811362/1902ac44-3eba-4697-b6f2-e962354f1b75)

**Comment by Snowflake6** [comment]
I just booted to find a notice of a new version for EP.  When I try installing I get the same virus warning.
![image](https://github.com/valinet/ExplorerPatcher/assets/5349216/f61c40cc-dbd5-4f2c-9698-ac6952cf587b)
![image](https://github.com/valinet/ExplorerPatcher/assets/5349216/c80f2642-24ec-4e98-9386-f8a429148aa6)

**Comment by Snowflake6** [comment]
Tried again and now its reporting another virus...
![image](https://github.com/valinet/ExplorerPatcher/assets/5349216/f124138d-1aa2-4eaf-98c3-0ab439e29147)

**Reply by prettyunusual** [reply]
I have been getting the same warning from Windows Security, is it a false positive or has the new ExplorerPatcher update been infected with Bladabindi!ml?

**Reply by pushih** [reply]
I also got the same warning when I tried to update to the latest ExplorerPatcher.

**Comment by turetafr** [comment]
I am also getting this warning. I am attempting to uninstall but explorer is very unhappy.

**Comment by y2k04** [comment]
https://github.com/valinet/ExplorerPatcher/issues/2873

**Comment by RedEd023** [comment]
I also got the issue this morning.

**Comment by ArXiLaMaS** [comment]
I got the same when I tried to update to the latest version but I get this Trojan:Win32/Wacatac.B!ml

![Screenshot 2024-02-21 182939](https://github.com/valinet/ExplorerPatcher/assets/106247976/f188d889-98c8-465f-b8ad-6fcb4146e9aa)

**Comment by jdpm1991** [comment]
Happened to me just now.

**Comment by Blunderfoot** [comment]
Same. issue.

**Comment by jumpott** [comment]
![image](https://github.com/valinet/ExplorerPatcher/assets/30414681/a28cc9eb-7faa-4c4d-8209-a3c0d148a429)

**Comment by gshock** [comment]
i got the same thing here. is there or should there be a way to verify the file hash of the download, or something? i love this tool, btw. it "fixed" my windows 11

**Comment by jw-mn** [comment]
My system also flagged it as having a backdoor control embedded.  Will need to stop updates if not resolved.

**Comment by blecaplol** [comment]
why are u doing this? u have such a nice program. why just why? fucking loser.

**Reply by lainverse** [reply]
This might be a false positive. Additionally, it's detected as having potentially unwanted behavior. Which is patching MS code, which it literally does. It's right in the name. So, options from least to highest probability are:
* author did something wrong;
* author got hacked and someone else uploaded a virus;
* MS is angry people don't like their W11 masterpiece of UI and flagged EP as PUP;
* Windows Defender made an oopsy and flagged ExplorerPatcher as a PUP since it literally uses behaviors similar to malicious programs to inject its code into Windows Explorer, so this has to happen sooner or later.

**Comment by vittoriop77** [comment]
Same here

**Comment by sonoks01** [comment]
I have the same issues. Please fix this.

Error message leads to KB of Microsoft:
https://www.microsoft.com/en-us/wdsi/threats/malware-encyclopedia-description?name=HackTool%3AWin64%2FPatcher&threatid=2147830899

![image](https://github.com/valinet/ExplorerPatcher/assets/66586752/50144cb0-5978-406a-9f60-1176fdd685eb)

**Reply by lainverse** [reply]
Try the latest release: [22621.3007.63.4](https://github.com/valinet/ExplorerPatcher/releases/tag/22621.3007.63.4_91aaa82)

**Comment by pyrates999** [comment]
All you can do is submit it as a false positive here: https://www.microsoft.com/en-us/wdsi/filesubmission

**Reply by jw-mn** [reply]
And what if the warning is not a false positive???

**Reply by pyrates999** [reply]
prove it.

**Reply by jw-mn** [reply]
pyrates999:  That's our point.  If there is not an issue, the project owner should speak up.  There are a lot of people that are using this and have been for some time with no issue.  We run AV for a reason and if something was accidentally committed to this most recent build, unintentionally, the author/owner should weigh in.  Maybe MS or other AV platforms have tweaked what they are looking for, maybe not.   Meanwhile, we stick with an older build until something is sorted out.  That is what reasonable people do.  I'm fine with the notice that this is not usually downloaded.  I'm NOT fine with the message that this file contains a backdoor for remote control.

**Reply by pyrates999** [reply]
The owner has.  This has happened before and it has always been a false positive.  The source code is available.  You can examine it and prove it.  You can do your own build as well.

**Reply by lainverse** [reply, marked as answer]
The author literally did so in the latest release:

> Note: This update has been flagged by Microsoft as HackTool:Win64/Patcher so we recommend to disable real-time protection before installing, and adding C:\Program Files\ExplorerPatcher into exclusions to prevent issues with uninstallation. We would like to assure you that such detections are false positive due to the nature of this software.

BTW, right now Windows Defender flags [22621.3007.63.3](https://github.com/valinet/ExplorerPatcher/releases/tag/22621.3007.63.3_1517de0) as PUP, but not latest [22621.3007.63.4](https://github.com/valinet/ExplorerPatcher/releases/tag/22621.3007.63.4_91aaa82)

**Reply by jw-mn** [reply]
lainverse:  good response!  What I was hoping for.

**Reply by DingleDangleNootNoot** [reply]
Holy carp there we go! I appreciate this as direct proof of what was said, I appreciate your comments here!

**Reply by jw-mn** [reply]
Latest release [22621.3007.63.4](https://github.com/valinet/ExplorerPatcher/releases/tag/22621.3007.63.4_91aaa82) installs with no warnings or issues.  Thanks for the tip - lainverse!

**Comment by ARK9975** [comment]
Just chiming in to say that I also have this problem.

**Comment by TotemicWeevil** [comment]
How many of you are using windows 10/11's in built AV and getting a report that there's something wrong?
Yeah.. me too....
UNTIL I installed a 3rd party firewall & AV and re-scanned. It came up clean.
If you can, try it and see what happens.

**Comment by lainverse** [comment]
Virustotal for [22621.3007.63.4](https://github.com/valinet/ExplorerPatcher/releases/tag/22621.3007.63.4_91aaa82):
https://www.virustotal.com/gui/file/d13c956afd1babccadeb2ecbb93506dd4f88bcc689dbffc9320f1d8e8f5a0dfa
When I opened it first it displayed 5 detections. It already went down to 4.
Virus total for [22621.3007.63.3](https://github.com/valinet/ExplorerPatcher/releases/tag/22621.3007.63.3_1517de0):
https://www.virustotal.com/gui/file/0d46bd6e83d661567efd6d79ae760a041f6a1ea72b4b043da428c7fbb93ad27f
This one did manage to trigger 8 false-positives including Microsoft (Windows Defender) HackTool:Win64/Patcher.
Virust total for [22621.3007.63.2](https://github.com/valinet/ExplorerPatcher/releases/tag/22621.3007.63.2_08dec82):
https://www.virustotal.com/gui/file/15a62dcf6c083fd9bdd183f1ec3c77269090aa9a91c7b15ce7d30ca7c0683183
Shows 3 detections.

So, that's just antivirus have false-positives as usual and slightly more than usual in case of 22621.3007.63.3.

**Reply by lainverse** [reply]
UPD: 63.4 dropped to 3, the same 3 as 63.2.

The most likely cause of false-positives in 63.3:
Setup: There should now be fewer .prev files, and uninstallation should be cleaner as well. (https://github.com/valinet/ExplorerPatcher/commit/296c6a0fed7e29797cace2fe7d3b873416130ee6)

And here is the reversed part in 63.4:
Setup: Reverted the method for ending explorer.exe and its subprocesses. (https://github.com/valinet/ExplorerPatcher/commit/fdc357b23fd2b8d612863f18e553543ad3b2c92c)

**Reply by Amrsatrio** [reply]
@lainverse

This is the VT for 62.2 release which has been up for two months:
https://www.virustotal.com/gui/file/e44790e25db09d1fdcaa1b4a8e868a31d646a260c9df4923aea7be8efa0d8e1d
2 detections.

This is the VT for 63.2 release (no text saying pre-release after its title): https://www.virustotal.com/gui/file/a25e2487bb4b638d6333d652db58532f3f29dd5ddb7711f70f52e0e61e8d3f51/detection
10 detections.

This is the VT for 63.3 release:
https://www.virustotal.com/gui/file/0d46bd6e83d661567efd6d79ae760a041f6a1ea72b4b043da428c7fbb93ad27f
8 detections.

63.4 prerelease:
https://www.virustotal.com/gui/file/d13c956afd1babccadeb2ecbb93506dd4f88bcc689dbffc9320f1d8e8f5a0dfa?nocache=1
3 detections, Defender doesn't detect this. (Update: Increased to 4, but that one is not a big name AV)

Honestly speaking, 63.2 and 63.3 began with only two detections just like 62.2. It only went up due to the exe being spread even more as I promote it into a release and brings people to that page to download the update. 63.4's detections, I *believe*, will definitely rise to the same degree as 63.2 and 63.3 if I promote it to release.

So yeah, guess we can't have fun. I'm putting 62.2 as the latest release for now, and 63.4 as the latest prerelease with latest fixes for those who would whine about broken stuff in 62.2 like notification center positioning. Other than that, everything works fine.
